### PR TITLE
Issue 56 feat/create user profile screen

### PR DESCRIPTION
**Pull request by N00BKU**

### CODE REVIEW

**Review by PoProstuMieciek — CHANGES_REQUESTED** [review]
It doesn't work for me. I am running local webserver and trying to access [localhost:3000/profile/test](http://localhost:3000/profile/test).
It throws me a bunch of errors, nothing is rendered on the page.

Logs: [localhost-1644308154541.log](https://github.com/coders-camp-2021-best-team/Freeverse/files/8021937/localhost-1644308154541.log)

File: src/components/molecules/UserInfo/UserInfo.jsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export const UserInfo = ({ userID, onClick, onPost }) => {
         const { profile_picture_url, displayName } = userData.data();
 
         return (
-            <div className={onPost ? '' : 'userinfo'}>
+            <div className={onPost ? '' : `user__info ${customClass}`}>
                 <ImageComponent
                     className='atom__image'
```
Comment (PoProstuMieciek) [inline src/components/molecules/UserInfo/UserInfo.jsx:24]: I think that you changed the class name for image in SCSS, but forgot to change it also in the JSX component

File: src/components/molecules/UserInfo/UserInfo.scss [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
 
-.userinfo {
+.user__info {
     order: 1;
 
     &__text {
         display: none;
     }
 
-    .atom__image {
```
Comment (PoProstuMieciek) [inline src/components/molecules/UserInfo/UserInfo.scss:11]: here, the thing that you changed

File: src/components/organisms/UserDetails/UserDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// import { Timestamp } from 'firebase/firestore'; TODAY'S DATE
```
Comment (PoProstuMieciek) [inline src/components/organisms/UserDetails/UserDetails.jsx:1]:
```js
import { Timestamp } from 'firebase/firestore';
import PropTypes from 'prop-types';
import { InformationRow, Text } from '../..';
import { useUserDetails } from '../../../hooks';
import { dateFormat } from '../../../utils';

import './UserDetails.scss';

export const UserDetails = ({ userID }) => {
    const { data: userData } = useUserDetails(userID);

    if (!userData?.data()) {
        return null;
    }

    const { birthday, hobbies, city } = userData.data();

    let birthday_str = dateFormat(birthday?.toDate(), 'DD.MM.YYYY');
    const age = Timestamp.now() - birthday;
    const years_old = Math.floor(age / 60 / 60 / 24 / 365);
    birthday_str += ` (${years_old} yo)`;

    const hobbies_str = hobbies.join(', ');

    return (
        <div>
            <Text size='medium'>User Details</Text>

            <InformationRow iconName='birthday'>
                {birthday_str || 'Not available'}
            </InformationRow>

            <InformationRow iconName='home'>
                {city || 'Not available'}
            </InformationRow>

            <InformationRow iconName='googleController'>
                {hobbies_str || 'Not available'}
            </InformationRow>
        </div>
    );
};

UserDetails.propTypes = {
    userID: PropTypes.string.isRequired
};
```

File: src/pages/Profile/Profile.jsx [hunk 4]
```diff
@@ -1,18 +1,38 @@
-import PropTypes from 'prop-types';
```
Comment (PoProstuMieciek) [inline src/pages/Profile/Profile.jsx:1]:
```js
import { useParams } from 'react-router';
import {
    ImageComponent,
    UserInfo,
    UserDetails,
    PostCollection,
    Text
} from '../../components';
import { useUserDetails } from '../../hooks';
import DEFAULT_BACKGROUND from '../../images/background.png';

import './Profile.scss';

export const ProfilePage = () => {
    const params = useParams();
    const userID = params.id;

    const { data: userData } = useUserDetails(userID);

    // TODO: wait for hooks at PR #82
    // const { data: postsData } = useUserPosts(userID);

    if (!userData?.data()) {
        return null;
    }

    const { background_picture_url } = userData.data();

    return (
        <div className='profile__page'>
            <div>
                <ImageComponent
                    customClass='background__image'
                    src={background_picture_url || DEFAULT_BACKGROUND}
                />
            </div>

            <UserInfo userID={userID} customClass='profile__page__user__info' />

            <UserDetails userID={userID} />

            <div>
                <Text size='medium'>Posts</Text>

                {/* TODO: posts */}
                <PostCollection userPosts={[]} />
            </div>
        </div>
    );
};
```

**Review by PoProstuMieciek — CHANGES_REQUESTED** [review]
You are using the hooks a bit wrong, but that's ok. I fixed some small bugs for you, because we have very little time before today's meeting.

**Review by PoProstuMieciek — APPROVED** [review]
I added a slight change into styles myself (we have no time, sorry) because `Post` component styles were broken on `ProfileScreen`, but it is OK now.

**Review by Mariusz-web — APPROVED** [review]

**Review by KonTra7cf — APPROVED** [review]

**Pull request merged**